diff --git a/node_modules/@serverless-stack/cli/bin/scripts.js b/node_modules/@serverless-stack/cli/bin/scripts.js
index c218515..0888467 100755
--- a/node_modules/@serverless-stack/cli/bin/scripts.js
+++ b/node_modules/@serverless-stack/cli/bin/scripts.js
@@ -435,8 +435,18 @@ async function run() {
 
       // Prepare app
       prepareCdk(argv, cliInfo, config)
-        .then(() => internals[script](argv, config, cliInfo))
-        .catch((e) => exitWithMessage(e.message));
+      .then(() => {
+        const res = internals[script](argv, config, cliInfo);
+        res.then(
+         fp.flow(
+           fp.map('status'),
+           fp.indexOf('failed'),
+           fp.tap(found => process.exit(found >= 0))
+         )
+        );
+        return res;
+      })
+      .catch((e) => exitWithMessage(e.message));
 
       break;
     }
